test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,644 @@
+        this.controllerService = new ControllerServiceImpl(controllerWrapper.getControllerService(),
+                GrpcAuthHelper.getDisabledAuthHelper(), requestTracker, true, 2);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        this.controller.close();
+        this.connectionPool.close();
+        this.controllerWrapper.close();
+        this.serverListener.close();
+        this.serviceBuilder.close();
+        this.zkTestServer.close();
+        this.keyValueTableFactory.close();
+    }
+
+    /**
+     * Create Stream and KVTable with same name under same scope.
+     */
+    @Test
+    public void testCreateStreamKVTable() throws ExecutionException, InterruptedException {
+        String newScope = "NewScope";
+        KeyValueTableInfo newKVT = newKeyValueTableName(newScope);
+        String streamName = newKVT.getKeyValueTableName();
+        int streamSegments = 1;
+        String readerGroupName = "readerGroup";
+        String readerName = "reader";
+        String routingKey = "routingKey";
+        String streamEvent = "Stream event";
+        int numEvents = 100;
+
+        final ScalingPolicy scalingPolicy = ScalingPolicy.fixed(streamSegments);
+        final StreamConfiguration config = StreamConfiguration.builder()
+                .scalingPolicy(scalingPolicy)
+                .build();
+
+        assertTrue("The requested Scope must get created", this.controller.createScope(newScope).join());
+
+        assertTrue("The requested Stream must get created", this.controller.createStream(newScope,
+                streamName, config).join());
+
+        assertTrue("The requested KVT must get created",
+                this.controller.createKeyValueTable(newScope, newKVT.getKeyValueTableName(),
+                        DEFAULT_CONFIG).join());
+
+        assertEquals("Table partition value should be same as passed", DEFAULT_CONFIG.getPartitionCount(),
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegmentsForKeyValueTable(newScope, newKVT.getKeyValueTableName()),
+                        RuntimeException::new).getSegments().size());
+
+        assertEquals("The stream segment count should be same as passed", streamSegments,
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegments(newScope, streamName),
+                        RuntimeException::new).getSegments().size());
+
+        // To read and write few sample events on the stream having same name as the KVTable.
+        readWriteOnStream(newScope, streamName, numEvents, routingKey, streamEvent, readerGroupName, readerName);
+
+        String keyFamily = "KeyFamily1";
+        // inserts and fetches few Table entries for validate basic functioning of KVTable
+        insertAndGetFromKVTable(newScope, newKVT, keyFamily);
+
+        // Cleans up stream resources by sealing existing stream segments & then deletes the stream.
+        cleanUpStream(newScope, streamName);
```

Comment:
Do we really need to do these? We run cluster cleanup after each test in `@after` so, IMO it is unnecessary to delete the scope and the streams, as each test is using its own cluster, and it may be reducing the tests' performance.